File: src/utils/cache.ts
import { redis } from '@/redis';

export async function getCachedContentUrl(key: string) {
  const result = await redis.get(key);
  return result || null;
}

export async function setCachedContentUrl(key: string, value: string) {
  await redis.setex(key, 60 * 60 * 12, value);
}

File: src/redis/cache.go
package redis

import (
	"context"
	"time"

	goredis "github.com/redis/go-redis/v9"
)

func GetCachedContentURL(ctx context.Context, key string) (string, error) {
	result, err := Client().Get(ctx, key).Result()
	if err != nil {
		if err == goredis.Nil {
			return "", nil
		}

		return "", err
	}

	return result, nil
}

func SetCachedContentURL(ctx context.Context, key, value string, ttl time.Duration) error {
	return Client().SetEx(ctx, key, value, ttl).Err()
}
